File: examples/eApi v1.9/java/eAPI-v19-connector/src/main/java/cz/monetplus/mips/eapi/v19/connector/entity/responses/GooglepayInitParams.java
package cz.monetplus.mips.eapi.v19.connector.entity.responses;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import cz.monetplus.mips.eapi.v19.connector.entity.SignBase;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@EqualsAndHashCode(callSuper = false)
@Data
@AllArgsConstructor @NoArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
class GooglepayInitParams extends SignBase {
    private int apiVersion = 2;
    private int apiVersionMinor = 0;
    private String paymentMethodType;
    private List<String> allowedCardNetworks = new ArrayList<>();
    private List<String> allowedCardAuthMethods = new ArrayList<>();
    private Boolean assuranceDetailsRequired;
    private Boolean billingAddressRequired;
    private String billingAddressParametersFormat;
    private String tokenizationSpecificationType;
    private String gateway;
    private String gatewayMerchantId;
    private String googlepayMerchantId;
    private String merchantName;
    private String environment;
    private String totalPriceStatus;
    private String countryCode;

    @Override
    public String toSign() {
        StringBuilder sb = new StringBuilder();
        add(sb, apiVersion);
        add(sb, apiVersionMinor);
        add(sb, paymentMethodType);
        for (String value : allowedCardNetworks) {
            add(sb, value);
        }
        for (String value : allowedCardAuthMethods) {
            add(sb, value);
        }
        add(sb, assuranceDetailsRequired);
        add(sb, billingAddressRequired);
        add(sb, billingAddressParametersFormat);
        add(sb, tokenizationSpecificationType);
        add(sb, gateway);
        add(sb, gatewayMerchantId);
        add(sb, googlepayMerchantId);
        add(sb, merchantName);
        add(sb, environment);
        add(sb, totalPriceStatus);
        add(sb, countryCode);
        deleteLast(sb);
        return sb.toString();
    }
}

File: examples/eApi v1.9/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/Response/GooglepayInitParams.cs
using System.Text;

namespace CsobGatewayClientExample.Communication.DataObjects;

public class GooglepayInitParams : SignBase
{
    private int ApiVersion { get; } = 2;

    private int ApiVersionMinor { get; } = 0;

    public string? PaymentMethodType { get; set; }

    // ReSharper disable once CollectionNeverUpdated.Global
    private List<string>? AllowedCardNetworks { get; set; }

    // ReSharper disable once CollectionNeverUpdated.Global
    private List<string>? AllowedCardAuthMethods { get; set; }


    public bool? AssuranceDetailsRequired { get; set; }

    public bool? BillingAddressRequired { get; set; }

    public string? BillingAddressParametersFormat { get; set; }

    public string? TokenizationSpecificationType { get; set; }

    public string? Gateway { get; set; }

    public string? GatewayMerchantId { get; set; }

    public string? GooglepayMerchantId { get; set; }

    public string? MerchantName { get; set; }

    public string? Environment { get; set; }

    public string? TotalPriceStatus { get; set; }

    public string? CountryCode { get; set; }

    public override string ToSign()
    {
        var sb = new StringBuilder();
        Add(sb, ApiVersion);
        Add(sb, ApiVersionMinor);
        if (AllowedCardNetworks != null)
            foreach (var value in AllowedCardNetworks)
                Add(sb, value);
        if (AllowedCardAuthMethods != null)
            foreach (var value in AllowedCardAuthMethods)
                Add(sb, value);
        Add(sb, GooglepayMerchantId);
        Add(sb, MerchantName);
        Add(sb, TotalPriceStatus);
        DeleteLast(sb);
        return sb.ToString();
    }
}